### PR TITLE
Network sync refactoring (part 2)

### PR DESCRIPTION
**Pull request by nazar-pc**
This is the second step towards #8686.

As https://github.com/paritytech/substrate/issues/8686#issuecomment-1045993814 suggests, this moves `api.v1` proto scheme and a bunch of data structures related to sync into new `sc-network-sync` module, which `sc-network` now depends on.

This is mostly an implementation detail of `sc-network`, most of `sc-network` public API is unchanged.

I also noticed that there are some shared data structures that will be used in `sc-network`, `sc-network-sync` and upcoming `sc-network-light` (will be sent in a separate PR) and moved them into another new crate `sc-network-common`. `sc-network-common` is fairly minimal and can be expanded more in the future if necessary.

Overall this is mostly moving code around and really there is no changes in business logic.

I have made changes in commits such that they should make sense and compile/run (except two where I intentionally moved files in specific way). If this is merged without squashing, the history of all files should be preserved throughout all of the changes (useful for tracing origin of code changes in the future).

Reviewing individual commits is recommended.

P.S. I was unlucky to do this right when libp2p was updated :upside_down_face:

### CODE REVIEW

**Comment by nazar-pc** [comment]
I see Polkadot digs a bit deeper into networking and fails CI because of that.

Would you prefer to add re-exports of `sc-network-common` in `sc-network` to fix that for now or create a companion that switches some of the things to `sc-network-common` already?

**Review by arkpar — APPROVED** [review]

File: client/network/sync/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[package]
+description = "Substrate network protocol"
```
Comment (bkchr) [inline client/network/sync/Cargo.toml:2]:
```suggestion
description = "Substrate sync network protocol"
```

File: client/network/src/config.rs [hunk 2]
```diff
@@ -21,12 +21,16 @@
 //! The [`Params`] struct is the struct that must be passed in order to initialize the networking.
 //! See the documentation of [`Params`].
 
-pub use crate::{
+// TODO: Below set of re-exports are for compatibility purposes for Polkadot, replace with regular
+//  imports at some point in the future once Polkadot is migrated
```
Comment (bkchr) [inline client/network/src/config.rs:25]:
```suggestion
```

**Review by bkchr — APPROVED** [review]
Ty!

**Comment by bkchr** [comment]
bot merge

**Comment by nazar-pc** [comment]
CI failure must be related to https://github.com/paritytech/substrate/pull/11340 that was just merged

**Pull request merged**